modules/asset-manager-workflowoperation/src/main/java/org/opencastproject/workflow/handler/assetmanager/AssetManagerStorageMoveOperationHandler.java

```diff
@@ -0,0 +1,106 @@
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.workflow.handler.assetmanager;
+
+import static org.opencastproject.util.data.Collections.smap;
+import static org.opencastproject.util.data.Tuple.tuple;
+
+import org.opencastproject.assetmanager.api.Version;
+import org.opencastproject.assetmanager.impl.TieredStorageAssetManagerJobProducer;
+import org.opencastproject.assetmanager.impl.VersionImpl;
+import org.opencastproject.job.api.Job;
+import org.opencastproject.job.api.JobContext;
+import org.opencastproject.job.api.JobImpl;
+import org.opencastproject.mediapackage.MediaPackage;
+import org.opencastproject.workflow.api.AbstractWorkflowOperationHandler;
+import org.opencastproject.workflow.api.WorkflowInstance;
+import org.opencastproject.workflow.api.WorkflowOperationException;
+import org.opencastproject.workflow.api.WorkflowOperationInstance;
+import org.opencastproject.workflow.api.WorkflowOperationResult;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.SortedMap;
+
+public class AssetManagerStorageMoveOperationHandler extends AbstractWorkflowOperationHandler {
+
+  private static final Logger logger = LoggerFactory.getLogger(AssetManagerStorageMoveOperationHandler.class);
+
+  /** The asset manager. */
+  private TieredStorageAssetManagerJobProducer tsamjp;
+
+  /** The configuration options for this handler */
+  private static final SortedMap<String, String> CONFIG_OPTIONS = smap(
+          tuple("target-storage", "The target storage type"),
+          tuple("target-version", "The specific version to move"));
+
+  @Override
+  public SortedMap<String, String> getConfigurationOptions() {
+    return CONFIG_OPTIONS;
+  }
+
+  /** OSGi DI */
+  public void setJobProducer(TieredStorageAssetManagerJobProducer tsamjp) {
+    this.tsamjp = tsamjp;
+  }
+
+
+  @Override
+  public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobContext context)
+          throws WorkflowOperationException {
+    MediaPackage mp = workflowInstance.getMediaPackage();
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+
+    logger.debug("Working on mediapackage {}", mp.getIdentifier().toString());
+
+    String targetStorage = StringUtils.trimToNull(operation.getConfiguration("target-storage"));
+    if (tsamjp.datastoreExists(targetStorage)) {
```

Comment:
Reverse test: if (!tsamjp.datastoreExists(targetStorage)). Trim string configuration just in case?

Reply:
Typo repaired!